File: backend/app/schemas/public.py
from pydantic import BaseModel, ConfigDict

from app.api.docs import CLUB_STATS_EXAMPLE


class ClubStats(BaseModel):
    model_config = ConfigDict(json_schema_extra={"example": CLUB_STATS_EXAMPLE})

    clients_count: int
    trainers_count: int
    classes_next_7_days: int
    active_subscriptions_count: int

File: frontend/src/shared/api/contracts/public.ts
import { z } from "zod";

export const clubStatsSchema = z.object({
  clients_count: z.number(),
  trainers_count: z.number(),
  classes_next_7_days: z.number(),
  active_subscriptions_count: z.number()
});

export type ClubStats = z.infer<typeof clubStatsSchema>;
